checks/raw/pinned_dependencies_test.go

```diff
@@ -843,6 +875,149 @@ func TestDockerfileInsecureDownloadsLineNumber(t *testing.T) {
 	}
 }
 
+func TestDockerfileWithHeredocsInsecureDownloadsLineNumber(t *testing.T) {
+	t.Parallel()
+	//nolint:govet
+	tests := []struct {
+		name             string
+		filename         string
+		processingErrors int
+		expected         []struct {
+			snippet   string
+			startLine uint
+			endLine   uint
+			pinned    bool
+			t         checker.DependencyUseType
+		}
+	}{
+		{
+			name:             "dockerfile heredoc downloads",
+			filename:         "./testdata/Dockerfile-download-heredoc",
+			processingErrors: 1,
```

Comment:
Is the python script necessary? I was a little confused at what the processing error was.
```
RUN <<EOT
#!/usr/bin/env python
print("hello world")
EOT
```

Reply:
If this is something only the heredoc would support, I can see why it would be here, but at the very least can we add a comment that the processing error is for the python script?

Reply:
The main point of the python test case is that she-bangs are allowed in here-docs (to specify another interpreter than a shell) and shouldn't cause a fatal error.

The mvdan/sh error is indeed slightly strange (the `foo` comes from [here](https://github.com/mvdan/sh/blob/v3.7.0/syntax/parser.go#L1809)).

I'll add a comment to the test case.